File: 3.无重复字符的最长子串/001.java
import java.io.LineNumberInputStream;

class Solution {
    public int lengthOfLongestSubstring(String s) {
        int base=0;
        int max = 0;
        int offset = 0;
        byte[] str = s.getBytes();
        main_loop:
        while(base+offset<s.length()){
            for(int i=0; i<offset; i++){
                if(str[base+i]==str[base+offset]){
                    max = max>offset ? max : offset;
                    offset--;
                    base++;
                    continue main_loop;
                }
            }
            offset++;
        }
        return max>offset?max:offset;
    }
}

File: 3.无重复字符的最长子串/001.c
int lengthOfLongestSubstring(char * s){
    int count = 0;
    int max = 0;
    int i;
    while(s[count]!='\0'){
        for(i=0; i<count; i++){
            if(s[i]==s[count]){
                max = count>max ? count : max;
                s++;
                count-=2;
                break;
            }
        }
        count++;
    }
    return max>count?max:count;
}

int main(int argv,char*args){
    char* s = "pwwkew";
    return lengthOfLongestSubstring(s);
}